File: src/main/java/Box/BoxTester.java
package Box;

import java.util.Scanner;

public class BoxTester {
    public static void main(String[] args) {

        float length, width;
        int height;

        Scanner scanner = new Scanner(System.in);

        System.out.print("請輸入長度: ");
        length = scanner.nextFloat();
        System.out.print("請輸入寬度: ");
        width = scanner.nextFloat();
        System.out.print("請輸入高度: ");
        height = scanner.nextInt();

        Box3 box3 = new Box3();
        Box5 box5 = new Box5();

        if (box3.validate(length, width, height)){
            System.out.println("可選擇Box3");
        }else if (box5.validate(length, width, height)){
            System.out.println("可選擇Box5");
        } else
        System.out.println("不符合容量大小！");

    }
}

File: src/main/kotlin/com/train/BoxTester.kt
package com.train

import java.util.*

fun main(args: Array<String>) {

    var length: Float
    var width: Float
    var height: Int

    var scanner = Scanner(System.`in`)

    print("請輸入長度: ")
    length = scanner.nextFloat()
    print("請輸入寬度: ")
    width = scanner.nextFloat()
    print("請輸入高度: ")
    height = scanner.nextInt()

    val box3Kt = Box3Kt()
    val box5Kt = Box5Kt()

    if (box3Kt.validate(length, width, height)){
        println("可選擇Box3")
    } else if (box5Kt.validate(length, width, height)){
        println("可選擇Box5")
    } else {
        println("不符合容量大小！")
    }
}

open class BoxKt(var length: Float, var width: Float, var height: Int) {
    open fun validate (length: Float, width: Float, height: Int): Boolean =
        (length <= this.length && width <= this.width && height <= this.height)
}

class Box3Kt: BoxKt(23f,14f,13)

class Box5Kt: BoxKt(39.5f, 27.5f, 23)